pandas/core/frame.py

```diff
@@ -4706,7 +4706,7 @@ def predicate(arr: ArrayLike) -> bool:
 
             return True
 
-        mgr = self._mgr._get_data_subset(predicate)
+        mgr = self._mgr._get_data_subset(predicate, copy=True)
```

Comment:
instead of adding a keyword in the Manager method could we just do a .copy after _get_data_subset?

Reply:
Yes, good point. I was originally thinking to add it here to ensure that the manager can be smarter to know what needs to be copied and what not. But since the predicate is based on the block dtype, this can never split any blocks, so I suppose there will never be such a case where the subset might already be a copy. So yes, doing a `.copy()` here is indeed much simpler.

There is one difference though: `combine` (used by `_get_data_subset`) doesn't do consolidation, while `copy()` does. So if your original dataframe is not fully consolidated, the current PR keeps more of the performance improvement of https://github.com/pandas-dev/pandas/pull/42611. Of course, at the cost of an unconsolidated return value and a potential later consolidation.

Either way is fine for me. Doing a simpler `copy()` will also keep it closer to the previous behaviour since `iloc` also consolidates I think.